enterprise/frontend/src/metabase-enterprise/advanced_permissions/components/RoleAttributeMappingModal/RoleAttributeMappingModal.tsx

```diff
@@ -0,0 +1,59 @@
+import React, { useCallback, useEffect } from "react";
+import { withRouter } from "react-router";
+import { useSelector } from "react-redux";
+import { push } from "react-router-redux";
+import { useDispatch } from "metabase/lib/redux";
+import { fetchAttributes } from "metabase-enterprise/sandboxes/actions";
+import { getAttributes } from "metabase-enterprise/sandboxes/selectors";
+import Databases from "metabase/entities/databases";
+import { State } from "metabase-types/store";
+import LoadingAndErrorWrapper from "metabase/components/LoadingAndErrorWrapper/LoadingAndErrorWrapper";
+import { RoleAttributeMappingModalParams } from "metabase-enterprise/advanced_permissions/types";
+import { getParentPath } from "metabase/hoc/ModalRoute";
+import RoleAttributeMappingModalView from "./RoleAttributeMappingModalView";
+
+interface RoleAttributeMappingModalProps {
+  params: RoleAttributeMappingModalParams;
+  route: {
+    path: string;
+  };
+}
+
+const RoleAttributeMappingModal = ({
+  params,
+  route,
+}: RoleAttributeMappingModalProps) => {
+  const databaseId = parseInt(params.impersonatedDatabaseId);
+  const attributes = useSelector(getAttributes);
+
+  const database = useSelector((state: State) =>
```

Comment:
Nit: you can import `useSelector` from `metabase/lib/redux` as well to avoid having to manually type state

I will try to get a PR up soon restricting imports of `useSelector` and `useDispatch` directly from `react-redux`